tests/fast-integration/skr-test/test.js

```diff
@@ -1,63 +1,67 @@
-const {KCPWrapper, KCPConfig} = require('../kcp/client');
+const {
+  gatherOptions,
+  withSuffix,
+  withInstanceID,
+  provisionSKRInstance,
+  director,
+  commerceMockTest,
+  oidcE2ETest,
+} = require('./index');
+const {getEnvOrThrow} = require('../utils');
+const {unregisterKymaFromCompass} = require('../compass');
+const {deprovisionSKRInstance} = require('./provision/deprovision-skr');
+const {
+  getSKRConfig,
+  prepareCompassResources,
+  initK8sConfig,
+} = require('./helpers');
 
-const kcp = new KCPWrapper(KCPConfig.fromEnv());
+const skipProvisioning = process.env.SKIP_PROVISIONING === 'true';
+const provisioningTimeout = 1000 * 60 * 30; // 30m
+const deprovisioningTimeout = 1000 * 60 * 95; // 95m
+let globalTimeout = 1000 * 60 * 70; // 70m
+const slowTime = 5000;
 
-describe('Execute SKR test', function() {
-  this.timeout(60 * 60 * 1000 * 3); // 3h
-  this.slow(5000);
+describe('SKR test', function() {
+  if (!skipProvisioning) {
+    globalTimeout += provisioningTimeout + deprovisioningTimeout;
+  }
+  this.timeout(globalTimeout);
+  this.slow(slowTime);
 
-  const provisioningTimeout = 1000 * 60 * 30; // 30m
-  const deprovisioningTimeout = 1000 * 60 * 95; // 95m
+  let options = gatherOptions(); // with default values
+  let shoot;
 
-  before('Provision SKR', async function() {
-    try {
-      this.options = gatherOptions();
-      console.log(`Provision SKR with instance ID ${this.options.instanceID}`);
-      const customParams = {
-        oidc: this.options.oidc0,
-      };
-      const skr = await provisionSKR(keb, kcp, gardener,
-          this.options.instanceID,
-          this.options.runtimeName,
-          null,
-          null,
-          customParams,
-          provisioningTimeout);
-
-      const runtimeStatus = await kcp.getRuntimeStatusOperations(this.options.instanceID);
-      console.log(`\nRuntime status after provisioning: ${runtimeStatus}`);
-
-      this.shoot = skr.shoot;
-      await addScenarioInCompass(director, this.options.scenarioName);
-      await assignRuntimeToScenario(director, this.shoot.compassID, this.options.scenarioName);
-      initializeK8sClient({kubeconfig: this.shoot.kubeconfig});
-    } catch (e) {
-      throw new Error(`before hook failed: ${e.toString()}`);
-    } finally {
-      const runtimeStatus = await kcp.getRuntimeStatusOperations(this.options.instanceID);
-      await kcp.reconcileInformationLog(runtimeStatus);
+  before('Ensure SKR is provisioned', async function() {
+    this.timeout(provisioningTimeout);
+    if (!skipProvisioning) {
+      shoot = await provisionSKRInstance(options, provisioningTimeout);
+    } else {
+      console.log('Gather information from externally provisioned SKR and prepare the compass resources');
+      const instanceID = getEnvOrThrow('INSTANCE_ID');
+      const suffix = getEnvOrThrow('TEST_SUFFIX');
+      options = gatherOptions(
+          withInstanceID(instanceID),
+          withSuffix(suffix),
+      );
```

Comment:
nit nit: IMHO it is easier to read then there is no **double negotiation** in the **first case**.
If you prefer the existing solution however, please just leave it as it is :)

```suggestion
    if (skipProvisioning) {
      console.log('Gather information from externally provisioned SKR and prepare the compass resources');
      const instanceID = getEnvOrThrow('INSTANCE_ID');
      const suffix = getEnvOrThrow('TEST_SUFFIX');
      options = gatherOptions(
          withInstanceID(instanceID),
          withSuffix(suffix),
      );
    } else {
      shoot = await provisionSKRInstance(options, provisioningTimeout);    
```

Reply:
makes sense, changed it